main/src/cgeo/geocaching/ui/Variable.java

```diff
@@ -0,0 +1,285 @@
+package cgeo.geocaching.ui;
+
+import android.content.Context;
+import android.content.res.Resources;
+import android.graphics.Color;
+import android.graphics.Typeface;
+import android.support.v4.content.ContextCompat;
+import android.support.v7.widget.GridLayout;
+import android.text.Editable;
+import android.text.InputType;
+import android.text.TextWatcher;
+import android.util.TypedValue;
+import android.view.Gravity;
+import android.widget.EditText;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import java.io.Serializable;
+import java.util.List;
+
+import cgeo.geocaching.R;
+
+/**
+ * Created by Slartibartfast on 30/06/2017.
+ *
+ * Class used to display a variable with an equation, such as:
+ *  X = a^2 = b^2
+ */
+
+public class Variable extends LinearLayout {
+
+    // Data used to capture the state of this Variable such that it can be restored again later.
+    public static class VariableData implements Serializable {
+        char name;
+        String expression; // Note, we have to use a String rather than an Editable as Editable's can't be serialized
+
+        public VariableData(char theName) {
+            name = theName;
+            expression = "";
+        }
+    }
+
+    VariableData mData;
```

Comment:
There are a couple of other 'm' prefixes out there but their rare so I'll go through and remove them all.
I've also done my best to go through and explicitly set all variables to 'private' unless otherwise indicated.